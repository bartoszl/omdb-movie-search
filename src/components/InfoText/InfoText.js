import React from 'react';
import PropTypes from 'prop-types';
import InfoLabel from '../InfoLabel';

const InfoText = ({ label, children }) => (
  <p>
    <InfoLabel>
      { label }
    </InfoLabel>
    { children }
  </p>
);

InfoText.propTypes = {
  label: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default InfoText;
