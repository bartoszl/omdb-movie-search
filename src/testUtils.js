import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import theme from './config/theme';

const TestProviders = ({ children }) => (
  <ThemeProvider theme={theme}>
    <MemoryRouter>
      {children}
    </MemoryRouter>
  </ThemeProvider>
);

TestProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: TestProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
