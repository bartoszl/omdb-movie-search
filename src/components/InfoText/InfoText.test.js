import React from 'react';
import { render, cleanup } from '../../testUtils';
import InfoText from './InfoText';

describe('InfoLabel', () => {
  const text = 'Text';
  const label = 'Label';

  afterEach(() => {
    cleanup();
  });

  test('matches snapshot', () => {
    const { asFragment } = render((
      <InfoText label={label}>
        { text }
      </InfoText>
    ));

    expect(asFragment()).toMatchSnapshot();
  });
});
