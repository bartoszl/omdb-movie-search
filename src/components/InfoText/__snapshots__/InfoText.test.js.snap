// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoLabel matches snapshot 1`] = `
<DocumentFragment>
  <p>
    <span
      class="sc-bdnylx iXUFvQ"
    >
      Label
    </span>
    Text
  </p>
</DocumentFragment>
`;
